service/src/main/java/uk/nhs/adaptors/oneoneone/cda/report/controller/utils/ReportParserUtil.java

```diff
@@ -13,35 +13,63 @@
 public class ReportParserUtil {
 
     private static final String MESSAGE_ID_NODE = "//*[local-name()='MessageID']";
+    private static final String ADDRESS_NODE = "//*[local-name()='Address']";
     private static final String DISTRIBUTION_ENVELOPE_NODE = "//*[local-name()='DistributionEnvelope']";
     private static final String HEADER_NODE = "//*[local-name()='header']";
 
-    public static Map<ReportElement, String> parseReportXml(String reportXml) throws DocumentException {
+    public static Map<ReportElement, String> parseReportXml(String reportXml) throws DocumentException, SoapClientException {
 
         Map<ReportElement, String> reportElementsMap = new HashMap<>();
 
         SAXReader reader = new SAXReader();
         Document document = reader.read(new StringReader(reportXml));
 
         reportElementsMap.put(ReportElement.MESSAGE_ID, getMessageId(document));
+        reportElementsMap.put(ReportElement.ADDRESS, getAddress(document));
         reportElementsMap.put(ReportElement.DISTRIBUTION_ENVELOPE, getDistributionEnvelope(document));
         reportElementsMap.put(ReportElement.TRACKING_ID, getTrackingId(document));
         return reportElementsMap;
     }
 
-    private static String getMessageId(Document document) {
+    private static String getMessageId(Document document) throws SoapClientException {
+        String messageId = "MessageId";
+        String errorLocation = "Header";
         Node messageIdNode = document.selectSingleNode(MESSAGE_ID_NODE);
+        try {
+            return messageIdNode.getText();
+        } catch (NullPointerException e) {
+            throw new SoapClientException(messageId, errorLocation, e);
+        }
+    }
+
+    private static String getAddress(Document document) {
+        String defaultToAddress = "http://www.w3.org/2005/08/addressing/anonymous";
```

Comment:
It's not this class's responsibility to know Error response default reply to address. This class should only know how to extract Address from SOAP.